bluepysnap/edges.py

```diff
@@ -27,8 +26,353 @@
 from bluepysnap.sonata_constants import DYNAMICS_PREFIX, Edge, ConstContainer
+from bluepysnap import utils
+
+
+class Edges:
+    """The top level Edges accessor."""
+
+    def __init__(self, circuit):
+        """Initialize the top level Edges accessor."""
+        self._circuit = circuit
+        self._config = self._circuit.config['networks']['edges']
+        self._populations = self._get_populations()
+
+    def _get_populations(self):
+        """Collect the different EdgePopulation."""
+        res = {}
+        for file_config in self._config:
+            storage = EdgeStorage(file_config, self._circuit)
+            for population in storage.population_names:  # pylint: disable=not-an-iterable
+                if population in res:
+                    raise BluepySnapError("Duplicated edge population: '%s'" % population)
+                res[population] = storage.population(population)
+        return res
+
+    @cached_property
+    def population_names(self):
+        """Returns all the population names from the Circuit."""
+        return sorted(self._populations)
+
+    @cached_property
+    def property_dtypes(self):
+        """Returns all the property dtypes for the Circuit."""
+
+        def _update(d, index, value):
+            if index not in d:
+                d[index] = value
+            elif d[index] != value:
+                raise BluepySnapError("Same property with different "
+                                      "dtype. {}: {}!= {}".format(index, value, d[index]))
+
+        res = dict()
+        for pop in self.populations():
+            for varname, dtype in pop.property_dtypes.iteritems():
+                _update(res, varname, dtype)
+        return pd.Series(res)
+
+    def keys(self):
+        """Returns iterator on the population names.
+
+        Made to simulate the behavior of a dict.keys().
+        """
+        return (name for name in self.population_names)
+
+    def values(self):
+        """Returns iterator on the EdgePopulations.
+
+        Made to simulate the behavior of a dict.values().
+        """
+        return (self[name] for name in self.population_names)
+
+    def items(self):
+        """Returns iterator on the tuples (population name, EdgePopulations).
+
+        Made to simulate the behavior of a dict.items().
+        """
+        return ((name, self[name]) for name in self.population_names)
+
+    # helper renaming
+    names = keys
+    populations = values
+
+    def __getitem__(self, population_name):
+        """Access the EdgePopulation corresponding to the population 'population_name'."""
+        try:
+            return self._populations[population_name]
+        except KeyError:
+            raise BluepySnapError("{} not a edge population.".format(population_name))
+
+    def __iter__(self):
+        """Allows iteration over the different EdgePopulation."""
+        return iter(self.names())
```

Comment:
Maybe it's better to use the original function name "keys" and not the alias ? 
Same for self.populations() below.

Reply:
yeah indeed